python/pylibcugraph/pylibcugraph/replication.pyx

```diff
@@ -0,0 +1,210 @@
+    cugraph_induced_subgraph_get_edge_type_ids,
+    cugraph_induced_subgraph_get_subgraph_offsets,
+    cugraph_induced_subgraph_result_free,
+)
+from pylibcugraph.resource_handle cimport (
+    ResourceHandle,
+)
+from pylibcugraph.utils cimport (
+    assert_success,
+    assert_CAI_type,
+    copy_to_cupy_array,
+    create_cugraph_type_erased_device_array_view_from_py_obj
+)
+
+
+def replication(ResourceHandle resource_handle,
+                src_array,
+                dst_array,
+                weight_array,
+                edge_id_array,
+                edge_type_id_array):
+    """
+        Compute vertex pairs that are two hops apart. The resulting pairs are
+        sorted before returning.
+
+        Parameters
+        ----------
+        resource_handle : ResourceHandle
+            Handle to the underlying device resources needed for referencing data
+            and running algorithms.
+        
+        src_array : device array type, optional
+            Device array containing the vertex identifiers of the source of each
+            directed edge. The order of the array corresponds to the ordering of the
+            dst_array, where the ith item in src_array and the ith item in dst_array
+            define the ith edge of the graph.
+        
+        dst_array : device array type, optional
+            Device array containing the vertex identifiers of the destination of
+            each directed edge. The order of the array corresponds to the ordering
+            of the src_array, where the ith item in src_array and the ith item in
+            dst_array define the ith edge of the graph.
+
+        weight_array : device array type, optional
+            Device array containing the weight values of each directed edge. The
+            order of the array corresponds to the ordering of the src_array and
+            dst_array arrays, where the ith item in weight_array is the weight value
+            of the ith edge of the graph.
+        
+        edge_id_array : device array type, optional
+            Device array containing the edge id values of each directed edge. The
+            order of the array corresponds to the ordering of the src_array and
+            dst_array arrays, where the ith item in edge_id_array is the id value
+            of the ith edge of the graph.
+        
+        edge_type_id_array : device array type, optional
+            Device array containing the edge type id values of each directed edge. The
+            order of the array corresponds to the ordering of the src_array and
+            dst_array arrays, where the ith item in edge_type_id_array is the type id
+            value of the ith edge of the graph.
+
+        Returns
+        -------
+        return a cupy arrays of 'src' and 'dst', 'weight', 'edge_id' and 'edge_type_id'
+        (if the were passed)
+    """
+
+
+    #print("src array = \n", src_array)
+    print("wgt array = \n", weight_array)
```

Comment:
Can you remove this and all other debug prints?

Reply:
Done.